crates/turbo-tasks-memory/src/scope.rs

```diff
@@ -136,6 +134,13 @@ pub struct TaskScopeState {
     /// All child scopes, when the scope becomes active, child scopes need to
     /// become active too
     children: CountHashSet<TaskScopeId, BuildNoHashHasher<TaskScopeId>>,
+    /// flag if this scope has unfinished tasks
+    has_unfinished_tasks: bool,
```

Comment:
It would be interesting to add a note here to explain how this is different from `TaskScope::unfinished_tasks`